From ec277f1056227689c8875d1c73602b47bfe72633 Mon Sep 17 00:00:00 2001
From: Neal Gompa <neal@gompa.dev>
Date: Mon, 28 Sep 2026 18:12:45 -0400
Subject: [PATCH] Revert "arm64: Limit memory allocations to 4GB boundary"

This causes failed memory allocations on some ARM-based platforms,
notably Apple Silicon Macs.

This reverts commit 1cae861314f43540cde773fab8b6ed769dc5f106.

Reviewed-by: Leo Sandoval <lsandova@redhat.com>
Signed-off-by: Neal Gompa <neal@gompa.dev>
Part-of: <https://gitlab.freedesktop.org/gnu-grub/grub/-/merge_requests/274>
Upstream: https://gitlab.freedesktop.org/gnu-grub/grub/-/commit/711110daccaf15df695b270d13f76b5bc84a64dd
Signed-off-by: Julien Olivain <ju.o@free.fr>
---
 include/grub/arm64/efi/memory.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/grub/arm64/efi/memory.h b/include/grub/arm64/efi/memory.h
index 2c64918e3..c6cb32417 100644
--- a/include/grub/arm64/efi/memory.h
+++ b/include/grub/arm64/efi/memory.h
@@ -1,6 +1,6 @@
 #ifndef GRUB_MEMORY_CPU_HEADER
 #include <grub/efi/memory.h>
 
-#define GRUB_EFI_MAX_USABLE_ADDRESS 0xffffffff
+#define GRUB_EFI_MAX_USABLE_ADDRESS 0xffffffffffffULL
 
 #endif /* ! GRUB_MEMORY_CPU_HEADER */
-- 
2.56.0

